Classes/BagBuilder/Abstract.php

```diff
@@ -169,6 +169,49 @@ public function getWhereClause()
         return implode(' AND ', $this->whereClauseParts);
     }
 
+    /**
+     * Returns a WHERE clause part for the bag to create.
+     *
+     * If the bag does not have such limitation imposed upon, the return value
+     * will be empty.
+     *
+     * @param string $key the limitation key to return, must not be empty
+     *
+     * @return string WHERE clause part for the bag to create, may be empty
+     */
+    public function getWhereClausePart($key)
```

Comment:
As seminars now requires PHP >= 7.0, please use a parameter type and return type declaration.